api/breaker/breaker_test.go

```diff
@@ -0,0 +1,550 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package breaker
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCircuitBreaker_generation(t *testing.T) {
+	t.Parallel()
+	clock := clockwork.NewFakeClock()
+
+	cb, err := New(Config{
+		Clock:    clock,
+		Interval: time.Second,
+		Trip:     StaticTripper(false),
+	})
+	require.NoError(t, err)
+
+	generation, state := cb.currentState(clock.Now().UTC())
+	require.Equal(t, uint64(1), generation)
+	require.Equal(t, StateStandby, state)
+	require.Equal(t, clock.Now().UTC().Add(time.Second), cb.expiry)
+
+	clock.Advance(500 * time.Millisecond)
+	generation, state = cb.currentState(clock.Now().UTC())
+	require.Equal(t, uint64(1), generation)
+	require.Equal(t, StateStandby, state)
+	clock.Advance(501 * time.Millisecond)
+	generation, state = cb.currentState(clock.Now().UTC())
+	require.Equal(t, uint64(2), generation)
+	require.Equal(t, StateStandby, state)
+	require.Equal(t, clock.Now().UTC().Add(time.Second), cb.expiry)
+
+	for i := 0; i < 1000; i++ {
```

Comment:
Is this meant to run 1000 times without advancing the clock?

Reply:
Yeah this is testing that generations will increase without adjusting the state, I suppose 1000 might be a bit excessive though.